lib/buffer.js

```diff
@@ -658,6 +658,87 @@ Buffer.prototype.includes = function includes(val, byteOffset, encoding) {
   return this.indexOf(val, byteOffset, encoding) !== -1;
 };
 
+Buffer.prototype.indexes = function indexes(match, limit) {
+  if (!Buffer.isBuffer(match))
+    throw new TypeError('argument must be a Buffer');
+  const indexes = [];
+  let i = -1;
+
+  let len = 0
+  while ((i = this.indexOf(match, i + 1)) != -1) {
+    indexes.push(i);
+    len++;
+    if (len >= limit)
+      break;
+  }
+  return indexes;
+};
+
+Buffer.prototype.split = function split(sep, encoding, limit) {
+  if (!limit && typeof encoding === 'number')
+    limit = encoding;
+  if (sep)
+    sep = Buffer.from(sep, encoding); // allows all constructor APIs
```

Comment:
unnecessary overhead. if `sep` is a buffer then `Buffer.from()` will make a copy of it. since we're not changing the values of `sep` there's no need for this.